### PR TITLE
Auto-reload for local webserver

### PR DESCRIPTION
**Pull request by sareensumanau**
/claim #78 

Updated PR.

### CODE REVIEW

**Comment by sareensumanau** [comment]
@dm161 The library is working as expected, on any change in file, its reloading the server.

I am facing one problem in connecting with postgres instance from air docker instance.

In our current approach, we are not using docker compose to start air and postgres instance, as they both are running unaware of each other, the air docker instance is having issues connecting with postgres docker instance exposed on host port of 9999.

I was able to ssh to docker container for air use the command to connect with database using host as `host.docker.internal`, but changing the config and using docker command directly is resulting in error

```2023/04/20 14:17:30 unable to connect to postgres: dial tcp 127.0.0.1:5432: connect: connection refused```

1. Do you see where I am making a mistake or missing something ?
2. Should we use docker-compose to setup air and postgres ?

File: local.env [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+EMAIL_API_KEY=a123
+PORT=9876
+JWT_SIGNING_KEY=a123
+SESSION_KEY=a123
+ADMIN_EMAIL=x@example.com
+SUPPORT_EMAIL=support@example.com
+MAILERLITE_API_KEY=a123
+SENTRY_DSN=https://localhost:123
+ENV=dev
+HEROKU_POSTGRESQL_PINK_URL="postgresql://postgres:passw0rd!@localhost:9999/postgres?sslmode=disable"
```
Comment (dm161) [inline local.env:19]:
```suggestion
HEROKU_POSTGRESQL_PINK_URL="postgresql://postgres:passw0rd!@host.docker.internal:9999/postgres?sslmode=disable"
```

i would try this?

File: run-local-webserver.sh [hunk 2]
```diff
@@ -1,50 +1,10 @@
...
-
-go build -o bin/server cmd/server/main.go
-
-./bin/server
+docker run -it --rm \
+    -v $(pwd):/app \
+    -w /app/cmd/server \
+    -e HOST=0.0.0.0 \
+    -e PORT=9876 \
+    --env-file local.env \
```
Comment (dm161) [inline run-local-webserver.sh:8]:
```suggestion
    --env-file local.env \
    --network=host \
```
or this

**Comment by sareensumanau** [comment]
@dm161 The issue was with postgres connection string if we pass it as a string then it won't work with docker. The trick is to take username, password, port etc seperately.

https://stackoverflow.com/questions/57696882/docker-golang-lib-pq-dial-tcp-127-0-0-15432-connect-connection-refused


The PR is working now and ready to be reviewed.

**Pull request merged**